pybliometrics/scopus/utils/startup.py

```diff
@@ -1,21 +1,81 @@
-import configparser
+from configparser import ConfigParser
 from collections import deque
 from pathlib import Path
+from typing import List, Dict, Optional, Type
+import os
+from warnings import warn
 
 from pybliometrics.scopus.utils.constants import CONFIG_FILE, RATELIMITS
 from pybliometrics.scopus.utils.create_config import create_config
 
-# Read/create config file (with fixture for RTFD.io)
-config = configparser.ConfigParser()
-config.optionxform = str
-try:
-    if not CONFIG_FILE.exists():
-        config = create_config()
+CUSTOM_DIR = None
+CUSTOM_KEYS = None
+
+def get_config() -> Type[ConfigParser]:
+    """Auxiliary function to get the config parser"""
+    # Read/create config file (with fixture for RTFD.io)
+    config = ConfigParser()
+    config.optionxform = str
+
+    if not CUSTOM_DIR and not CONFIG_FILE.exists():
+        warn('Please create a configuration file by initializing Scopus with init().\n'
+             'For more information visit: '
+             'https://pybliometrics.readthedocs.io/en/stable/configuration.html')
+    else:
+        if CUSTOM_DIR:
+            config.read(CUSTOM_DIR)
+        else:
+            warn('Please initialize Scopus with init().\n'
+                 'For more information visit: '
+                 'https://pybliometrics.readthedocs.io/en/stable/configuration.html', FutureWarning)
+            config.read(CONFIG_FILE)
+
+    return config
+
+def get_config_path() -> Type[Path]:
+    """Auxiliary function to get the configuration file path"""
+    return CUSTOM_DIR or CONFIG_FILE
+
+def get_keys() -> List[str]:
+    """Auxiliary function to get the API keys"""
+    if CUSTOM_KEYS:
+        keys = CUSTOM_KEYS
     else:
-        config.read(CONFIG_FILE)
-    KEYS = [k.strip() for k in config.get('Authentication', 'APIKey').split(",")]
-except EOFError:
-    pass
+        config = get_config()
+        keys = [k.strip() for k in config.get('Authentication', 'APIKey').split(",")]
+    return keys
+
+def get_throttling_params() -> Dict:
+    """Auxiliary function to get the throttling params"""
+    return {k: deque(maxlen=v) for k, v in RATELIMITS.items()}
+
```

Comment:
Why is it necessary to have a separate function for that?

Reply:
Putting the throttling_params inside a function has two advantages over having it as global code:
1. Better readability
2. The function will always return the current state in case something changes while execution.

Overall I prefer a avoiding using global code because the variables are static after the initial import. Encapsulating code in functions ensure updated returned values.